metrics/engine/ingester.go

```diff
@@ -87,6 +91,31 @@ func (oi *outputIngester) flushMetrics() {
 				oi.metricsEngine.markObserved(sm.Metric)
 				sm.Metric.Sink.Add(sample)
 			}
+
+			if _, ok := oi.seenTimeSeries[sample.TimeSeries]; !ok {
+				oi.seenTimeSeries[sample.TimeSeries] = struct{}{}
+			}
 		}
 	}
+
+	oi.warnTimeSeriesLimit()
+}
+
+func (oi *outputIngester) warnTimeSeriesLimit() {
+	if len(oi.seenTimeSeries) <= oi.timeSeriesLimit {
+		return
+	}
```

Comment:
Can you move this check outside of the method?

I.e. this method should only be called if the limit was exceeded. This pattern of immediately returning avoids repeating the same check if the function is called in several places, but it has the bigger drawback of not being clear when reading the calling code. The reader needs to be aware that this is _maybe_ logging a warning.